src/command/dev/next/router/binary.rs

```diff
@@ -134,13 +136,25 @@ where
                 "info".to_string(),
                 "--dev".to_string(),
             ];
-            let mut env = HashMap::from_iter([("APOLLO_ROVER".to_string(), "true".to_string())]);
+
+            // We set the APOLLO_KEY here, but it might be overriden by RemoteRouterConfig. That
+            // struct takes the who_am_i service, gets an identity, and checks whether the
+            // associated API key (if present) is of a graph-level actor; if it is, we overwrite
+            // the env key with it because we know it's associated with the target graph_ref and
+            let mut env = HashMap::from_iter([
+                ("APOLLO_ROVER".to_string(), "true".to_string()),
+                ("APOLLO_KEY".to_string(), self.credential.api_key.clone()),
+            ]);
+
+            // See note above on env creation
+            if let Some(api_key) = remote_config.as_ref().and_then(|c| c.api_key().clone()) {
+                env.insert("APOLLO_KEY".to_string(), api_key);
+            }
+
```

Comment:
minor stylistic comment, no action needed.

I would consider composing the APOLLO_KEY in a contiguous statement.

```
// beautiful descriptive comment about how APOLLO_KEY is computed as you already have above.

// bad pseudo code to demonstrate computing the api_key 
// in a way that is clearer to Nik
let apollo_key_computed = match remote_config.as_ref() {
     Some(_) => api_key,
     None => self.credential.api_key.clone()
}

let mut env = HashMap::from_iter([
                ("APOLLO_ROVER".to_string(), "true".to_string()),
                ("APOLLO_KEY".to_string(), apollo_key_computed),
  ]);


            ```

Reply:
ahh, that's a good idea! adding it this way to reduce the overall number of matches:

```
            let api_key =
                if let Some(api_key) = remote_config.as_ref().and_then(|c| c.api_key().clone()) {
                    env.insert("APOLLO_KEY".to_string(), api_key);
                } else {
                    self.credential.api_key.clone()
                };

            let mut env = HashMap::from_iter([
                ("APOLLO_ROVER".to_string(), "true".to_string()),
                ("APOLLO_KEY".to_string(), api_key),
            ]);
```